sql/internal/migrate/entc.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2021-present The Atlas Authors. All rights reserved.
+// This source code is licensed under the Apache 2.0 license found
+// in the LICENSE file in the root directory of this source tree.
+
+//go:build ignore
+
+package main
+
+import (
+	"log"
+
+	"entgo.io/ent/entc"
+	"entgo.io/ent/entc/gen"
+)
+
+const (
+	initTpl = `// Client is the client that holds all ent builders.
+type Client struct {
+	config
+	{{- range $n := $.Nodes }}
+		// {{ $n.Name }} is the client for interacting with the {{ $n.Name }} builders.
+		{{ $n.Name }} *{{ $n.Name }}Client
+	{{- end }}
+	{{- template "client/fields/additional" $ }}
+	{{- with $tmpls := matchTemplate "client/fields/additional/*" }}
+		{{- range $tmpl := $tmpls }}
+			{{- xtemplate $tmpl $ }}
+		{{- end }}
+	{{- end }}
+}
+
+// NewClient creates a new client configured with the given options.
+func NewClient(opts ...Option) *Client {
+	cfg := config{log: log.Println, hooks: &hooks{}}
+	cfg.options(opts...)
+	client := &Client{config: cfg}
+	client.init()
+	return client
+}
+
+func (c *Client) init() {
+	{{- range $n := $.Nodes }}
+    	c.{{ $n.Name }} = New{{ $n.Name }}Client(c.config)
+	{{- end }}
+}`
+	testTpl = `{{ with extend $ "Package" "enttest" -}}
+	{{ template "header" . }}
+{{ end }}
+
+// Overridden by a custom template because it is unused in this project.`
```

Comment:
What do you think about creating a feature-flag for enttest (that is enabled by default) and disabling it here? I can do it.